ThePortugueseMan.MathGame Upload

ThePortugueseMan opened the pull request.

ThePortugueseMan commented:
Final  ThePortugueseMan.MathGame Upload

TheCSharpAcademy commented on ThePortugueseMan.MathGame/MathGame/Program.cs:
```diff
@@ -0,0 +1,360 @@
+    else { return aux / max_time; }
+}
+
+bool Sum (int _min, int _max)
```
Something is not quite right here. One of the terms of the sum is always the maximum number and you want it to be random..  (I've set the max to 200)
![image](https://user-images.githubusercontent.com/108711476/218338373-d22acaa6-7e36-413d-a327-78a240563ae4.png)

ThePortugueseMan replied:
Fixed. I was generating random number between "max" and "max+1" instead of "min" and "max+1"

TheCSharpAcademy commented on ThePortugueseMan.MathGame/MathGame/Program.cs:
```diff
@@ -0,0 +1,360 @@
+    }
+
+    return false;
+
```
Please remove unnecessary empty line. I'll add it to the automatic tests, but basically you don't want to have empty lines between closing brackets.

ThePortugueseMan replied:
Removed

TheCSharpAcademy commented on ThePortugueseMan.MathGame/MathGame/Program.cs:
```diff
@@ -0,0 +1,360 @@
+                Console.WriteLine("Invalid character!");
+                break;
+        }
+
```
Please remove unnecessary empty line 228.

ThePortugueseMan replied:
Removed

TheCSharpAcademy commented on ThePortugueseMan.MathGame/MathGame/Program.cs:
```diff
@@ -0,0 +1,360 @@
+
+void Settings(ref int min, ref int max, ref int numberOfRounds)
+{
+
```
Please remove unnecessary empty line 234

ThePortugueseMan replied:
Removed

TheCSharpAcademy commented on ThePortugueseMan.MathGame/MathGame/Program.cs:
```diff
@@ -0,0 +1,360 @@
+int min = 1;
+int max = 100;
+string? mode;
+
```
Please remove unnecessary empty lines 290,291

TheCSharpAcademy commented on ThePortugueseMan.MathGame/MathGame/Program.cs:
```diff
@@ -0,0 +1,360 @@
+        default:
+            break;
+    }
+
```
Please remove unnecessary empty line 315

TheCSharpAcademy commented on ThePortugueseMan.MathGame/MathGame/Program.cs:
```diff
@@ -0,0 +1,360 @@
+    Console.ReadLine();
+    Console.Clear();
+    return;
+
```
Please remove unnecessary empty lines 336 and 338

TheCSharpAcademy commented on ThePortugueseMan.MathGame/MathGame/Program.cs:
```diff
@@ -0,0 +1,360 @@
+        {
+            return true;
+        }
+
```
Please remove unnecessary empty line 39

TheCSharpAcademy commented on ThePortugueseMan.MathGame/MathGame/Program.cs:
```diff
@@ -0,0 +1,360 @@
+    int number2 = random.Next(number1, _max + 1);
+    int subResult = number2 - number1;
+    int answer;
+
```
Please remove unnecessary empty line 52

TheCSharpAcademy commented on ThePortugueseMan.MathGame/MathGame/Program.cs:
```diff
@@ -0,0 +1,360 @@
+    {
+        number1 = random.Next(_min, _max+1);
+        number2 = random.Next(number1+1, _max+1);
+        divResult = number2 / number1;
```
Looks like "divresult" isn't being used? If not it can be removed.

TheCSharpAcademy commented on ThePortugueseMan.MathGame/MathGame/Program.cs:
```diff
@@ -0,0 +1,360 @@
+        {
+            return true;
+        }
+
```
Please remove unnecessary empty line 111

TheCSharpAcademy commented on ThePortugueseMan.MathGame/MathGame/Program.cs:
```diff
@@ -0,0 +1,360 @@
+void GameSelect(int _numberOfRounds, int _min, int _max, List<Game> _games)
+{
+    int gameScore;
+    
```
Please remove unnecessary empty line 188

TheCSharpAcademy commented on ThePortugueseMan.MathGame/MathGame/Program.cs:
```diff
@@ -0,0 +1,360 @@
+
+        switch (gameMode = Console.ReadLine())
+        {
+            
```
Please remove unnecessary empty line 199

TheCSharpAcademy reviewed (changes requested):
Good work! Your code is well organized and everything worked well. I like how you solved the challenges and made the score based on the timer. Most comments are related to spacing (I'll make those automatic soon). I think there's just one bug with the Sums. Well done!

ThePortugueseMan commented:
Thank you so much for the feedback! I removed the spaces and fixed the bug!

I really appreciate what you are doing! Is there any way I could help? I
have some university level experience in programming.

The C# Academy ***@***.***> escreveu no dia domingo,
12/02/2023 à(s) 21:44:

> ***@***.**** requested changes on this pull request.
>
> Good work! Your code is well organized and everything worked well. I like
> how you solved the challenges and made the score based on the timer. Most
> comments are related to spacing (I'll make those automatic soon). I think
> there's just one bug with the Sums. Well done!
> ------------------------------
>
> In ThePortugueseMan.MathGame/MathGame/Program.cs
> <https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103879210>
> :
>
> > +}
> +
> +int CalculateScore (Stopwatch timeToAnswer)
> +{
> +    int max_time = 5000;
> +    int aux = 100000;
> +
> +    if (timeToAnswer.ElapsedMilliseconds<max_time)
> +    {
> +        int returner = aux / (int)timeToAnswer.ElapsedMilliseconds;
> +        return returner;
> +    }
> +    else { return aux / max_time; }
> +}
> +
> +bool Sum (int _min, int _max)
>
> Something is not quite right here. One of the terms of the sum is always
> the maximum number and you want it to be random.. (I've set the max to 200)
> [image: image]
> <https://user-images.githubusercontent.com/108711476/218338373-d22acaa6-7e36-413d-a327-78a240563ae4.png>
> ------------------------------
>
> In ThePortugueseMan.MathGame/MathGame/Program.cs
> <https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880200>
> :
>
> > +    int answer;
> +
> +
> +    Console.WriteLine($"{number2} - {number1} = ?");
> +    if (int.TryParse(Console.ReadLine(), out answer))
> +    {
> +        if (answer == subResult)
> +        {
> +            return true;
> +        }
> +
> +        else { return false; }
> +    }
> +
> +    return false;
> +
>
> Please remove unnecessary empty line. I'll add it to the automatic tests,
> but basically you don't want to have empty lines between closing brackets.
> ------------------------------
>
> In ThePortugueseMan.MathGame/MathGame/Program.cs
> <https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880242>
> :
>
> > +            case "m":
> +                gameScore = Game("MULTIPLICATION", _numberOfRounds, _min, _max);
> +                _games.Add(new Game("Multiplication", gameScore));
> +                break;
> +            case "r":
> +                gameScore = Game("RANDOM", _numberOfRounds, _min, _max);
> +                _games.Add(new Game("Random", gameScore));
> +                break;
> +            case "e":
> +                Console.Clear();
> +                return;
> +            default:
> +                Console.WriteLine("Invalid character!");
> +                break;
> +        }
> +
>
> Please remove unnecessary empty line 228.
> ------------------------------
>
> In ThePortugueseMan.MathGame/MathGame/Program.cs
> <https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880253>
> :
>
> > +                _games.Add(new Game("Random", gameScore));
> +                break;
> +            case "e":
> +                Console.Clear();
> +                return;
> +            default:
> +                Console.WriteLine("Invalid character!");
> +                break;
> +        }
> +
> +    }
> +}
> +
> +void Settings(ref int min, ref int max, ref int numberOfRounds)
> +{
> +
>
> Please remove unnecessary empty line 234
> ------------------------------
>
> In ThePortugueseMan.MathGame/MathGame/Program.cs
> <https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880282>
> :
>
> > +        Console.WriteLine("--------\n");
> +        Console.WriteLine();
> +        Console.Write("Enter the new value:\t");
> +        input = Console.ReadLine();
> +    }
> +    while (!(int.TryParse(input, out _)));
> +
> +    _ = int.TryParse(input, out valueToChange);
> +}
> +
> +List<Game> games = new List<Game>();
> +int numberOfRounds = 5;
> +int min = 1;
> +int max = 100;
> +string? mode;
> +
>
> Please remove unnecessary empty lines 290,291
> ------------------------------
>
> In ThePortugueseMan.MathGame/MathGame/Program.cs
> <https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880322>
> :
>
> > +        case "n":
> +            GameSelect(numberOfRounds, min, max, games);
> +            Console.Clear();
> +            break;
> +        case "s":
> +            Settings(ref min, ref max, ref numberOfRounds);
> +            Console.Clear();
> +            break;
> +        case "v":
> +            ViewPreviousGames(games);
> +            Console.Clear();
> +            break;
> +        default:
> +            break;
> +    }
> +
>
> Please remove unnecessary empty line 315
> ------------------------------
>
> In ThePortugueseMan.MathGame/MathGame/Program.cs
> <https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880408>
> :
>
> > +{
> +    Console.Clear();
> +    Console.WriteLine("PREVIOUS GAMES");
> +    Console.WriteLine("--------------");
> +    Console.WriteLine("Game Type\t\tScore");
> +
> +    for (int i = _games.Count(); i > 0; i--)
> +    {
> +        Console.WriteLine($"{_games[i - 1].gameType}\t\t\t{_games[i - 1].score}");
> +    }
> +
> +    Console.WriteLine("\n Press any key to return to the menu");
> +    Console.ReadLine();
> +    Console.Clear();
> +    return;
> +
>
> Please remove unnecessary empty lines 336 and 338
> ------------------------------
>
> In ThePortugueseMan.MathGame/MathGame/Program.cs
> <https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880486>
> :
>
> > +bool Sum (int _min, int _max)
> +{
> +    Random random= new Random();
> +    int number1 = random.Next(_min, _max + 1);
> +    int number2 = random.Next(_max, _max + 1);
> +    int answer;
> +
> +    Console.WriteLine($"{number1} + {number2} = ?");
> +
> +    if (int.TryParse(Console.ReadLine(), out answer))
> +    {
> +        if (answer == number1 + number2)
> +        {
> +            return true;
> +        }
> +
>
> Please remove unnecessary empty line 39
> ------------------------------
>
> In ThePortugueseMan.MathGame/MathGame/Program.cs
> <https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880496>
> :
>
> > +            return true;
> +        }
> +
> +    }
> +
> +    return false;
> +}
> +
> +bool Subtraction(int _min, int _max)
> +{
> +    Random random = new Random();
> +    int number1 = random.Next(_min, _max + 1);
> +    int number2 = random.Next(number1, _max + 1);
> +    int subResult = number2 - number1;
> +    int answer;
> +
>
> Please remove unnecessary empty line 52
> ------------------------------
>
> In ThePortugueseMan.MathGame/MathGame/Program.cs
> <https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880657>
> :
>
> > +
> +}
> +
> +bool Division( int _min, int _max)
> +{
> +    Random random = new Random();
> +    double divResult;
> +    int number1;
> +    int number2;
> +    int answer;
> +
> +    do
> +    {
> +        number1 = random.Next(_min, _max+1);
> +        number2 = random.Next(number1+1, _max+1);
> +        divResult = number2 / number1;
>
> Looks like "divresult" isn't being used? If not it can be removed.
> ------------------------------
>
> In ThePortugueseMan.MathGame/MathGame/Program.cs
> <https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880676>
> :
>
> > +bool Multiplication(int _min, int _max)
> +{
> +    Random random = new Random();
> +    int number1 = random.Next(_min, _max + 1);
> +    int number2 = random.Next(_min, _max + 1);
> +    int answer;
> +
> +    Console.WriteLine($"{number1} * {number2} = ?");
> +
> +    if (int.TryParse(Console.ReadLine(), out answer))
> +    {
> +        if (answer == number1 * number2)
> +        {
> +            return true;
> +        }
> +
>
> Please remove unnecessary empty line 111
> ------------------------------
>
> In ThePortugueseMan.MathGame/MathGame/Program.cs
> <https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880711>
> :
>
> > +                timeToAnswer.Stop();
> +                break;
> +        }
> +
> +        if (right) { score = score + CalculateScore(timeToAnswer); }
> +
> +        _numberOfRounds--;
> +    }
> +    DisplayScore(score);
> +    return score;
> +}
> +
> +void GameSelect(int _numberOfRounds, int _min, int _max, List<Game> _games)
> +{
> +    int gameScore;
> +
>
> Please remove unnecessary empty line 188
> ------------------------------
>
> In ThePortugueseMan.MathGame/MathGame/Program.cs
> <https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880729>
> :
>
> > +
> +void GameSelect(int _numberOfRounds, int _min, int _max, List<Game> _games)
> +{
> +    int gameScore;
> +
> +
> +    string? gameMode;
> +
> +    Console.Clear();
> +
> +    while(true) {
> +        Console.WriteLine("Select an operation to start\n\t- S - sum\n\t- s - subtraction\n\t- d - division\n\t- m - multiplication\n\t- r - random\n\ne - to exit to the menu\n");
> +
> +        switch (gameMode = Console.ReadLine())
> +        {
> +
>
> Please remove unnecessary empty line 199
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#pullrequestreview-1294787076>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APOM5LBZYHGQ5TYAYUFBL3DWXFKT7ANCNFSM6AAAAAAUYTPXDA>
> .
> You are receiving this because you authored the thread.Message ID:
> <TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13/review/1294787076@
> github.com>
>

TheCSharpAcademy commented:
> Thank you so much for the feedback! I removed the spaces and fixed the bug! I really appreciate what you are doing! Is there any way I could help? I have some university level experience in programming. The C# Academy ***@***.***> escreveu no dia domingo, 12/02/2023 à(s) 21:44:
> […](#)
> ***@***.**** requested changes on this pull request. Good work! Your code is well organized and everything worked well. I like how you solved the challenges and made the score based on the timer. Most comments are related to spacing (I'll make those automatic soon). I think there's just one bug with the Sums. Well done! ------------------------------ In ThePortugueseMan.MathGame/MathGame/Program.cs <[#13 (comment)](https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103879210)> : > +} + +int CalculateScore (Stopwatch timeToAnswer) +{ + int max_time = 5000; + int aux = 100000; + + if (timeToAnswer.ElapsedMilliseconds<max_time) + { + int returner = aux / (int)timeToAnswer.ElapsedMilliseconds; + return returner; + } + else { return aux / max_time; } +} + +bool Sum (int _min, int _max) Something is not quite right here. One of the terms of the sum is always the maximum number and you want it to be random.. (I've set the max to 200) [image: image] <https://user-images.githubusercontent.com/108711476/218338373-d22acaa6-7e36-413d-a327-78a240563ae4.png> ------------------------------ In ThePortugueseMan.MathGame/MathGame/Program.cs <[#13 (comment)](https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880200)> : > + int answer; + + + Console.WriteLine($"{number2} - {number1} = ?"); + if (int.TryParse(Console.ReadLine(), out answer)) + { + if (answer == subResult) + { + return true; + } + + else { return false; } + } + + return false; + Please remove unnecessary empty line. I'll add it to the automatic tests, but basically you don't want to have empty lines between closing brackets. ------------------------------ In ThePortugueseMan.MathGame/MathGame/Program.cs <[#13 (comment)](https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880242)> : > + case "m": + gameScore = Game("MULTIPLICATION", _numberOfRounds, _min, _max); + _games.Add(new Game("Multiplication", gameScore)); + break; + case "r": + gameScore = Game("RANDOM", _numberOfRounds, _min, _max); + _games.Add(new Game("Random", gameScore)); + break; + case "e": + Console.Clear(); + return; + default: + Console.WriteLine("Invalid character!"); + break; + } + Please remove unnecessary empty line 228. ------------------------------ In ThePortugueseMan.MathGame/MathGame/Program.cs <[#13 (comment)](https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880253)> : > + _games.Add(new Game("Random", gameScore)); + break; + case "e": + Console.Clear(); + return; + default: + Console.WriteLine("Invalid character!"); + break; + } + + } +} + +void Settings(ref int min, ref int max, ref int numberOfRounds) +{ + Please remove unnecessary empty line 234 ------------------------------ In ThePortugueseMan.MathGame/MathGame/Program.cs <[#13 (comment)](https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880282)> : > + Console.WriteLine("--------\n"); + Console.WriteLine(); + Console.Write("Enter the new value:\t"); + input = Console.ReadLine(); + } + while (!(int.TryParse(input, out _))); + + _ = int.TryParse(input, out valueToChange); +} + +List<Game> games = new List<Game>(); +int numberOfRounds = 5; +int min = 1; +int max = 100; +string? mode; + Please remove unnecessary empty lines 290,291 ------------------------------ In ThePortugueseMan.MathGame/MathGame/Program.cs <[#13 (comment)](https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880322)> : > + case "n": + GameSelect(numberOfRounds, min, max, games); + Console.Clear(); + break; + case "s": + Settings(ref min, ref max, ref numberOfRounds); + Console.Clear(); + break; + case "v": + ViewPreviousGames(games); + Console.Clear(); + break; + default: + break; + } + Please remove unnecessary empty line 315 ------------------------------ In ThePortugueseMan.MathGame/MathGame/Program.cs <[#13 (comment)](https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880408)> : > +{ + Console.Clear(); + Console.WriteLine("PREVIOUS GAMES"); + Console.WriteLine("--------------"); + Console.WriteLine("Game Type\t\tScore"); + + for (int i = _games.Count(); i > 0; i--) + { + Console.WriteLine($"{_games[i - 1].gameType}\t\t\t{_games[i - 1].score}"); + } + + Console.WriteLine("\n Press any key to return to the menu"); + Console.ReadLine(); + Console.Clear(); + return; + Please remove unnecessary empty lines 336 and 338 ------------------------------ In ThePortugueseMan.MathGame/MathGame/Program.cs <[#13 (comment)](https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880486)> : > +bool Sum (int _min, int _max) +{ + Random random= new Random(); + int number1 = random.Next(_min, _max + 1); + int number2 = random.Next(_max, _max + 1); + int answer; + + Console.WriteLine($"{number1} + {number2} = ?"); + + if (int.TryParse(Console.ReadLine(), out answer)) + { + if (answer == number1 + number2) + { + return true; + } + Please remove unnecessary empty line 39 ------------------------------ In ThePortugueseMan.MathGame/MathGame/Program.cs <[#13 (comment)](https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880496)> : > + return true; + } + + } + + return false; +} + +bool Subtraction(int _min, int _max) +{ + Random random = new Random(); + int number1 = random.Next(_min, _max + 1); + int number2 = random.Next(number1, _max + 1); + int subResult = number2 - number1; + int answer; + Please remove unnecessary empty line 52 ------------------------------ In ThePortugueseMan.MathGame/MathGame/Program.cs <[#13 (comment)](https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880657)> : > + +} + +bool Division( int _min, int _max) +{ + Random random = new Random(); + double divResult; + int number1; + int number2; + int answer; + + do + { + number1 = random.Next(_min, _max+1); + number2 = random.Next(number1+1, _max+1); + divResult = number2 / number1; Looks like "divresult" isn't being used? If not it can be removed. ------------------------------ In ThePortugueseMan.MathGame/MathGame/Program.cs <[#13 (comment)](https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880676)> : > +bool Multiplication(int _min, int _max) +{ + Random random = new Random(); + int number1 = random.Next(_min, _max + 1); + int number2 = random.Next(_min, _max + 1); + int answer; + + Console.WriteLine($"{number1} * {number2} = ?"); + + if (int.TryParse(Console.ReadLine(), out answer)) + { + if (answer == number1 * number2) + { + return true; + } + Please remove unnecessary empty line 111 ------------------------------ In ThePortugueseMan.MathGame/MathGame/Program.cs <[#13 (comment)](https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880711)> : > + timeToAnswer.Stop(); + break; + } + + if (right) { score = score + CalculateScore(timeToAnswer); } + + _numberOfRounds--; + } + DisplayScore(score); + return score; +} + +void GameSelect(int _numberOfRounds, int _min, int _max, List<Game> _games) +{ + int gameScore; + Please remove unnecessary empty line 188 ------------------------------ In ThePortugueseMan.MathGame/MathGame/Program.cs <[#13 (comment)](https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#discussion_r1103880729)> : > + +void GameSelect(int _numberOfRounds, int _min, int _max, List<Game> _games) +{ + int gameScore; + + + string? gameMode; + + Console.Clear(); + + while(true) { + Console.WriteLine("Select an operation to start\n\t- S - sum\n\t- s - subtraction\n\t- d - division\n\t- m - multiplication\n\t- r - random\n\ne - to exit to the menu\n"); + + switch (gameMode = Console.ReadLine()) + { + Please remove unnecessary empty line 199 — Reply to this email directly, view it on GitHub <[#13 (review)](https://github.com/TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13#pullrequestreview-1294787076)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APOM5LBZYHGQ5TYAYUFBL3DWXFKT7ANCNFSM6AAAAAAUYTPXDA> . You are receiving this because you authored the thread.Message ID: <TheCSharpAcademy/CodeReviews.Console.MathGame/pull/13/review/1294787076@ github.com>

The next feature of the platform is the ability to review code from belts below yours. It should be out in a few weeks. The best way you can help by now is recommending the C# academy on reddit whenever someone asks where to learn, if you think we deserve it. 😁 There's excellent material out there for beginners but nothing offering projects and especially code reviews. But of course, only do this if you think we deserve it :D
